src/palace/manager/api/controller/loan.py

```diff
@@ -480,6 +481,14 @@ def revoke(self, license_pool_id: int) -> OPDSEntryResponse | ProblemDetail:
             )
 
         work = pool.work
+        if not work:
+            # Somehow we just revoked a loan or hold for a LicensePool
+            # that has no Work. This shouldn't happen.
+            self.log.error(
+                "Revoked loan or hold for LicensePool %r which has no Work!",
+                pool,
+            )
+            return NOT_FOUND_ON_REMOTE
```

Comment:
It wasn't the case in this instance, but this could cause the same error, so I also added some error handling for `None` here as well.